101-recovery-services-create-vault-enable-diagnostics/azuredeploy.json

```diff
@@ -37,7 +37,7 @@
       "type": "Microsoft.OperationalInsights/workspaces",
       "apiVersion": "2015-11-01-preview",
       "name": "[variables('omsWorkspaceName')]",
-      "location": "East US",
+      "location": "[parameters('location')]",

```

Comment:
Microsoft.OperationalInsights/workspaces is not globally available so you need to use allowedValues here (for both clouds)